EIPS/eip-7886.md

```diff
@@ -0,0 +1,399 @@
+---
+eip: 7886
+title: Delayed execution
+description: Make blocks statically verifiable by charging coinbase for inclusion costs upfront
+author: Francesco D`Amato (@fradamt), Toni Wahrstätter (@nerolation)
+discussions-to: https://ethereum-magicians.org/t/eip-7886-delayed-execution/22890
+status: Draft
+type: Standards Track
+category: Core
+created: 2025-02-18
+---
+
+## Abstract
+
+This proposal makes (execution) blocks statically verifiable through minimal checks that only require the previous state, but no execution of the block's transactions, allowing validators attest to a block's validity without completing its execution. We allow transactions to be skipped when invalid at execution time, without invalidating the whole block. To ensure that even skipped transactions pay for their resources, the `COINBASE` pays for all inclusion costs upfront (base cost, calldata and blobs), and recovers the costs only when transactions are successfully executed.
+
+## Motivation
+
+A key advantage of this proposal is that it enables **asynchronous block validation**. Currently, blocks must be fully executed before validators can attest to their validity. This requirement creates a bottleneck in the consensus process, as attestors must wait for execution results before committing their votes.
+
+By introducing a mechanism where invalid transactions are skipped rather than invalidating the entire block, execution is no longer an immediate requirement for validation. Instead, a block’s validity can be determined based on its structural correctness and the upfront payment of inclusion costs by the `COINBASE`. This allows attestation to happen earlier, independent of execution, potentially enabling higher block gas limits.
+
+## Specification
+
+### Change to the header structure
+
+#### Deferred fields 
+
+In order to be verifiable before execution, the header cannot commit to any execution output. In particular, we need to defer these fields: `state_root, receipt_root, bloom, gas_used, requests_hash`. We replace them with the equivalent execution outputs from the parent block. 
+
+#### Coinbase signature over the header
+
+We include a signature from `COINBASE` over the rest of the header in the header , so that the `COINBASE` address can authorize the upfront payment of inclusion costs.
+
+The final header structure then is: 
+
+```python
+class Header:
```

Comment:
`Deferred` comment on the field is a little bit confusing given that these `parent_*` fields are validated immediately